source/Octopus.Tentacle.Tests.Integration/ClientScriptExecutionScriptServiceV2IsNotRetried.cs

```diff
@@ -0,0 +1,231 @@
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut;
+using NUnit.Framework;
+using Octopus.Tentacle.CommonTestUtils.Builders;
+using Octopus.Tentacle.Contracts;
+using Octopus.Tentacle.Contracts.ScriptServiceV2;
+using Octopus.Tentacle.Tests.Integration.Support;
+using Octopus.Tentacle.Tests.Integration.Util;
+using Octopus.Tentacle.Tests.Integration.Util.Builders;
+using Octopus.Tentacle.Tests.Integration.Util.Builders.Decorators;
+using Octopus.Tentacle.Tests.Integration.Util.TcpTentacleHelpers;
+
+namespace Octopus.Tentacle.Tests.Integration
+{
+    [IntegrationTestTimeout]
+    public class ClientScriptExecutionScriptServiceV2IsNotRetried : IntegrationTest
+    {
+        [Test]
+        [TestCaseSource(typeof(TentacleTypesToTest))]
+        public async Task WhenANetworkFailureOccurs_DuringStartScript_WithRetriesDisabled_TheCallIsNotRetried(TentacleType tentacleType)
+        {
+            using var clientTentacle = await new ClientAndTentacleBuilder(tentacleType)
+                .WithRetriesDisabled()
+                .WithPortForwarderDataLogging()
+                .WithResponseMessageTcpKiller(out var responseMessageTcpKiller)
+                .WithRetryDuration(TimeSpan.FromMinutes(4))
+                .WithTentacleServiceDecorator(new TentacleServiceDecoratorBuilder()
+                    .LogCallsToScriptServiceV2()
+                    .RecordExceptionThrownInScriptServiceV2(out var scriptServiceExceptions)
+                    .CountCallsToScriptServiceV2(out var scriptServiceCallCounts)
+                    .DecorateScriptServiceV2With(new ScriptServiceV2DecoratorBuilder()
+                        .BeforeStartScript(() =>
+                        {
+                            if (scriptServiceExceptions.StartScriptLatestException == null)
+                            {
+                                responseMessageTcpKiller.KillConnectionOnNextResponse();
+                            }
+                        })
+                        .Build())
+                    .Build())
+                .Build(CancellationToken);
+
+            var waitForFile = Path.Combine(clientTentacle.TemporaryDirectory.DirectoryPath, "waitforme");
+
+            var startScriptCommand = new StartScriptCommandV2Builder()
+                .WithIsolation(ScriptIsolationLevel.FullIsolation)
+                .WithMutexName("bob")
+                .WithScriptBody(new ScriptBuilder()
+                    .Print("hello")
+                    .WaitForFileToExist(waitForFile)
+                    .Print("AllDone"))
+                .Build();
+
+            List<ProcessOutput> logs = new List<ProcessOutput>();
+            Assert.ThrowsAsync<HalibutClientException>(async () => await clientTentacle.TentacleClient.ExecuteScriptAssumingException(startScriptCommand, logs, CancellationToken));
+
+            // Let the script finish.
+            File.WriteAllText(waitForFile, "");
```

Comment:
Because we are failing the first call and not retrying the script will never start so the File lock logic can be removed

Reply:
`.WaitForFileToExist(waitForFile)`

and

`File.WriteAllText(waitForFile, "");`

Reply:
It looks like the first call to start script is failed when the response comes backs, which would mean the script is running. Right?

Reply:
Yes good point, but we don't need the script to wait for anything though as the response is failed and we should never retry